framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1506,21 +1581,87 @@ public static Range getRange(AnnotationMirror rangeAnno) {
         return new ArrayList<>(boolSet);
     }
 
+    public Integer getMinLenValue(AnnotatedTypeMirror atm) {
+        return getMinLenValue(atm.getAnnotationInHierarchy(UNKNOWNVAL));
+    }
+
+    /**
+     * Used to find the minimum length of an array, which is useful for array bounds checking.
+     * Returns null there is no minimum length known, or if the passed annotation is null. Note that
+     * this routine handles actual {@link MinLen} annotations, because it is called by {@link
+     * ValueAnnotatedTypeFactory#aliasedAnnotation(AnnotationMirror)}, which transforms {@link
+     * MinLen} annotations into {@link ArrayLenRange} annotations.
+     */
+    public Integer getMinLenValue(AnnotationMirror annotation) {
+        if (annotation == null) {
+            return null;
+        }
+        if (AnnotationUtils.areSameByClass(annotation, MinLen.class)) {
+            return AnnotationUtils.getElementValue(annotation, "value", Integer.class, true);
+        } else if (AnnotationUtils.areSameByClass(annotation, ArrayLenRange.class)) {
+            return Long.valueOf(getRange(annotation).from).intValue();
+        } else if (AnnotationUtils.areSameByClass(annotation, ArrayLen.class)) {
+            return Collections.min(getArrayLength(annotation));
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Returns the minimum value of an integral type. This is used by the Index Checker's array
+     * bounds checking routines when determining the minimum length of an array from the annotation
+     * on the length of the array.
+     */
+    public Integer getMinLenValueFromLengthType(AnnotatedTypeMirror atm) {
```

Comment:
Why `LengthType`?  Currently this is only used for array lengths, but it is (or should be) a general purpose routine that other checkers could use.  I would name it something like `getMinIntegralValue` and remove the mention of the Index Checker from the Javadoc.

Oh maybe it's `LengthType` because this method returns an Integer instead of a Long?  In either case, this should return a long and then the caller of the method can downcast as needed.